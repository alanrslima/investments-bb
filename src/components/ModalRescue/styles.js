import { StyleSheet } from 'react-native';
import { colors, fonts } from '../../styles';

export default StyleSheet.create({
  container: {
    backgroundColor: colors.WHITE,
  },
  content: {
    alignItems: 'center',
    paddingHorizontal: 20,
    paddingBottom: 50,
    marginTop: 30,
  },
  title: {
    fontSize: fonts.HEADLINE,
    color: colors.BLACK,
    fontWeight: '600'
  },
  description: {
    textAlign: 'center',
    marginTop: 10,
    color: colors.BLUE,
    fontSize: fonts.DESCRIPTION
  }
});
