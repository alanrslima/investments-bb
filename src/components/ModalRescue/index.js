import React from 'react';
import { View, Text } from 'react-native';
import Modal from 'react-native-modal';

import styles from './styles';
import Button from '../Button';

export default function ModalRescue({ isVisible }) {
  return (
    <Modal
      backdropOpacity={0.3}
      isVisible={isVisible}
    >
      <View style={styles.container}>
        <View style={styles.content}>
          <Text style={styles.title}>RESGATE EFETUADO!</Text>
          <Text style={styles.description}>
            O valor solicitado estará em sua conta em até 5 dias úteis!
          </Text>
        </View>
        <Button
          text="NOVO RESGATE"
          onPress={() => { }}
        />
      </View>
    </Modal>
  );
}
