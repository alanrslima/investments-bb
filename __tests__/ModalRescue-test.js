import React from 'react';
import renderer from 'react-test-renderer';
import ModalRescue from '../src/components/ModalRescue';


test('renders correctly', () => {
  const tree = renderer.create(<ModalRescue
    isVisible
    onCancel={() => { }}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
