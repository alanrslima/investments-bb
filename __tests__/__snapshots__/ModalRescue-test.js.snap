// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "marginTop": 30,
            "paddingBottom": 50,
            "paddingHorizontal": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": 22,
              "fontWeight": "600",
            }
          }
        >
          RESGATE EFETUADO!
        </Text>
        <Text
          style={
            Object {
              "color": "#225AA0",
              "fontSize": 15,
              "marginTop": 10,
              "textAlign": "center",
            }
          }
        >
          O valor solicitado estará em sua conta em até 5 dias úteis!
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F6E155",
            "justifyContent": "center",
            "opacity": 1,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#225AA0",
              "fontSize": 15,
              "fontWeight": "bold",
            }
          }
        >
          NOVO RESGATE
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
